src/OpenRiaServices.Client/Test/Client.Test/Data/OperationTests.cs

```diff
@@ -439,6 +442,91 @@ public void Bug706066_CancelInCallback()
             EnqueueTestComplete();
         }
 
+
+
+        class ExceptionHandleSynchronizationContext : SynchronizationContext
+        {
+            public Exception LastException { get; set; }
+            public override void Send(SendOrPostCallback d, object state)
+            {
+                if (state is ExceptionDispatchInfo exceptionDispatchInfo)
+                {
+                    LastException = exceptionDispatchInfo.SourceException;
+                }
+
+                base.Send(d, state);
+            }
+        }
+
+
+
+
+        /// <summary>
+        /// When operation complated callback occurr exception,
+        /// check if an exception was thrown on the SynchronizationContext.
+        /// </summary>
+        [TestMethod]
+        [Asynchronous]
+        public void Operation_CompleteCallbackHasError_ThrowSycCtx()
+        {
+            TestDataContext ctxt = new TestDataContext(new Uri(TestURIs.RootURI, "TestDomainServices-TestCatalog1.svc"));
+            var query = ctxt.CreateQuery<Product>("GetProducts", null, false, true);
+
+            var syncCtx = new ExceptionHandleSynchronizationContext();
+            bool isCallbackCalled = false;
+            Exception callbackException = new Exception("callbackException");
+
+            Action<LoadOperation<Product>> callback = (op) =>
+            {
+                SynchronizationContext.SetSynchronizationContext(syncCtx);
+
```

Comment:
I am a bit unsure if you want to store `SynchronizationContext.Current` and "restore" it at the end of the test method, I don't remeber if there is any special SynchronizationContext setup for the tests (need to search for calls to SetSynchronizationContext, but if tests work then it should be fine)